pkg/scanners/terraform/parser/evaluator.go

```diff
@@ -240,15 +240,7 @@ func (e *evaluator) expandBlockForEaches(blocks terraform.Blocks) terraform.Bloc
 		if !forEachAttr.Value().IsNull() && forEachAttr.Value().IsKnown() && forEachAttr.IsIterable() {
 			var clones []cty.Value
 			_ = forEachAttr.Each(func(key cty.Value, val cty.Value) {
-
-				index := key
-
-				switch val.Type() {
-				case cty.String, cty.Number:
-					index = val
-				}
```

Comment:
why is the check for `val.Type()` made here previously? If the type isn't string or Number, should we still assign the key?

What if we did this instead?
```go
switch val.Type() {
				case cty.String, cty.Number:
					index = key
				}
```

Reply:
I have improved the type checking for the argument: according to the documentation it can be [set of strings or a map](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each#the-for_each-meta-argument). 

I also fixed a bug: when a resource contained a for-each, it was added to the context variables as a tuple instead of a map with [each object's keys](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each#the-each-object) as keys, so that when the resource was used as a for-each argument, the key was the tuple index (see https://github.com/aquasecurity/trivy-iac/pull/48/files#diff-737632faf6acfadc173b3898bc0f74abee40ece7522382231b53cb8909b87780R1004)

Reply:
great!